### DISCUSSION TITLE
Add users/contributors map to the dashboard

### DISCUSSION THREAD

**Discussion by a0m0rajab** [open-sauced/feedback#7] [Feature request]
due to the conversation in this thread at discord: https://discord.com/channels/714698561081704529/928693344358514698/threads/1102364402583339180

I wanted to know if this is a feature that should be implemented for open sauced. 
The map could be a heat map like this one: 
![image](https://github.com/open-sauced/feedback/assets/18273833/a87dc282-f7ac-4572-bc37-7a70bbfa560d)

or could be a pin map like this: 
![image](https://github.com/open-sauced/feedback/assets/18273833/e948e4a1-9742-443c-9e55-14fd4eb85a88)
https://githubcampus.expert/

For the technical details we have the next: 
- GitHub is not using standard location information
- Some users doe snot have map information 

The action is generating the heat map: https://github.com/tunaitis/contributor-map is using a database to map the users with the related country and has a normalisation function for some type of the locations. 

OpenSauced insights is showing the users logins/usernames which we can use for the call to GitHub GraphQL to get the location and use them, the issue here is that I do not think we can do it on the frontend on realtime without a need to have a database check on the backend. 

As for the pin map, it's using leaflet which requires the latitude and longitude of the users.

Here is another library that we can use for the heatmap though: https://datamaps.github.io/

**Comment by bdougie** [comment]
I think if this could be put together as a proof of concept in a separate repo, we could evaluate it. You provide a few different options that I'd love to see as a solution. 

I would lean towards using the https://github.com/open-sauced/api.opensauced.pizza to support this.

**Reply by a0m0rajab** [reply]
Do you like to have a proof of concept with heatmaps or pinpoints? 
I prefer the heatmap as a first step.

**Reply by bdougie** [reply]
Proof of concept would be ideal. Whichever option you'd prefer.

**Reply by a0m0rajab** [reply]
I will create a repo to test it. Will take couple of days (hopefully).

**Reply by a0m0rajab** [reply]
@bdougie I just wrote a simple CLI for that: https://github.com/a0m0rajab/contributor_map_npm 
this is the initial version, I could have few improvements though. 


Thank you for showing interest in such idea though!
